plugins/hypervisors/xenserver/src/main/java/com/cloud/hypervisor/xenserver/resource/CitrixResourceBase.java

```diff
@@ -1507,13 +1507,13 @@ public void destroyPatchVbd(final Connection conn, final Set<VM> vms) throws Xml
         final VDI patchVDI = findPatchIsoVDI(conn, sr);
         for (final VM vm : vms) {
             final String vmName = vm.getNameLabel(conn);
-            try {
-                if (!vmName.startsWith("r-") && !vmName.startsWith("s-") && !vmName.startsWith("v-")) {
-                    return;
-                }
-                final Set<VBD> vbds = vm.getVBDs(conn);
-                for (final VBD vbd : vbds) {
-                    if (Types.VbdType.CD.equals(vbd.getType(conn))) {
+            if (!vmName.startsWith("r-") && !vmName.startsWith("s-") && !vmName.startsWith("v-")) {
```

Comment:
To simplify the validation, we could use `StringUtils.startsWithAny`, from `commons.lang3`:

```java
if (!StringUtils.startsWithAny(vmName, "r-", "s-", v-")) {
```

Reply:
I would hesitate to make any further changes at this point. Feel free to raise a PR.